### ISSUE TITLE
Batch with prepared queries

### ISSUE THREAD

**Issue by sboo** [duoshuo/php-cassandra#40]
Whenever I'm trying to use prepared queries in a batch, I get this error:

```
Exception in Error.php line 123:
Server error: java.lang.IndexOutOfBoundsException: index: 28, length: 749941 (expected: range(0, 137))
```

Here's what I'm executing:

```
$batchRequest = new Cassandra\Request\Batch(null, null, [
        'names_for_values' => true,
 ]);
$preparedData = $connection->prepare('INSERT INTO table (qid, rid, value, tsr, tsa)
VALUES (:qid,:rid,:value,:tsr,:tsa)');

$values = [
    'qid' => '1',
    'rid' => '1',
    'value' => '55',
    'tsr' => 1427225763000,
    'tsa' => 1427225763000,
];

$batchRequest->appendQueryId($preparedData['id'],
            Cassandra\Request\Request::strictTypeValues($values, $preparedData['metadata']['columns']));

$connection->syncRequest($batchRequest);
```

Am I missing something or is there an issue?

**Comment by shen2** [comment]
Have you  tried to use a general request?

**Comment by sboo** [comment]
Yeah, with normal queries it works as expected. I tried to debug the issue, but I don't quite understand the details yet of the protocol

**Comment by steambao** [comment]
I tried this, and can confirm 'Server error: java.lang.IndexOutOfBoundsException' occurs when using "names_for_values = true" for batch requests.

**Comment by shen2** [comment]
@steambao The code is strictly obey the protocol.  It seems like a bug of Cassandra.
Can you have a try to find a way to avoid this problem?

**Comment by steambao** [comment]
Had a look today. Besides turning off names for values, not sure how or why the error. Looks according to spec as far as I can tell.

**Comment by markharding** [comment]
This used to work in v0.5.2 (https://packagist.org/packages/duoshuo/php-cassandra)... and if i revert back to this version the request works fine. Any ideas of what changed since then to cause the bug?

**Comment by markharding** [comment]
This became broken in commit 79a046de17a983b9f43cdfb21b8ac31fb2109ebe. It works in 8268f9af0ed9d4faf5a1c71a79680ec345e98c93

**Comment by firnas** [comment]
Line [Request:122](https://github.com/duoshuo/php-cassandra/blob/master/src/Request/Request.php#L122) causes the exception, (tested in cassandra enviroments 2.1.2 and 2.1.4). It makes no sense because the code looks correct according to specification.

**Comment by markharding** [comment]
I wonder if binary value could be affected by string case (ie. upper or lower)?

**Comment by withings-sas** [comment]
Hi, we have the same issue, but it does not work in v0.5.2 too: it looks like it works but in fact "names_for_values" is ignore in v0.5.2 so if you give the args in the good orders, it works.. But if you force names_for_values the same error occurs:

Caught exception: Server error: java.lang.IndexOutOfBoundsException: index: 241, length: 681061 (expected: range(0, 409))

**Comment by withings-sas** [comment]
In fact names_for_values never worked for batch queries, I tested several version and all returns the same error

**Comment by shen2** [comment]
Hi, all.
It seems to be a bug of Cassandra.
So, the best solution is to throw exception when use NAMES_FOR_VALUES flag in batch request.
see https://github.com/duoshuo/php-cassandra/commit/e8f94d2da2a7848aa2199d6fefac9292b339bca6
